lib/portlayer/storage/image_cache.go

```diff
@@ -232,6 +232,18 @@ func (c *NameLookupCache) WriteImage(op trace.Operation, parent *Image, ID strin
 	return i, nil
 }
 
+func (c *NameLookupCache) Read(op trace.Operation, store *url.URL, child, ancestor string, pathspec map[string]string, data bool) (io.ReadCloser, error) {
+	_, err := util.ImageStoreName(store)
+	if err != nil {
+		return nil, err
+	}
+
+	if archiver, ok := c.DataStore.(Archiver); ok {
+		return archiver.Read(op, store, child, ancestor, pathspec, data)
```

Comment:
`store` is a `url.URL`, and can be a volume store or an image store as far as the interface is concerned. That being said, both volume and image stores use the provided url to make sure the corresponding store actually exists.